projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_plugin.py

```diff
@@ -63,6 +78,37 @@ def vdk_configure(config_builder: ConfigurationBuilder) -> None:
     def vdk_command_line(root_command: click.Group):
         root_command.add_command(impala_query)
 
+    # the purpose of the below hook is to get reference to any registered lineage loggers by other plugins
+    @hookimpl(
+        trylast=True
+    )  # trylast because we want to have any lineage loggers already initialized
+    def vdk_initialize(self, context: CoreContext) -> None:
+        self._lineage_logger = context.state.get(LINEAGE_LOGGER_KEY)
+
+    # the purpose of the below hook is to calculate lineage information after
+    @hookimpl(hookwrapper=True)
+    def db_connection_execute_operation(
+        self, execution_cursor: ExecutionCursor
+    ) -> Optional[int]:
+        yield  # let the query execute first
+        try:
+            if (
+                self._lineage_logger
+            ):  # calculate and send lineage data only if there is a registered lineage logger
+                hive_cursor = cast(HiveServer2Cursor, execution_cursor)
+                lineage_data = get_lineage_data(hive_cursor)
+                if (
+                    lineage_data
+                ):  # some queries do not have data lineage - select 1, create table, compute stats, etc
```

Comment:
Do we still request a profile in case of `select 1`, and if yes - what's the overhead? For example, how many select 1 is the driver sending per minute, and does it make sense to exclude such `select 1` beforehand, to reduce the pressure on Impala?

Reply:
Yep we request a profile for select 1 queries. The overhead should be minimal, the profile is already calculated by the query engine. The amount of select 1 queries is the same as the actual queries being executed.
It is a good suggestion to exclude them as it should be fairly straightforward to do so. Will update the implementation,